validator/client/src/main/java/tech/pegasys/teku/validator/client/restapi/apis/DeleteKeys.java

```diff
@@ -49,14 +47,21 @@ public DeleteKeys() {
             .response(SC_OK, "Success response", ValidatorTypes.DELETE_KEYS_RESPONSE_TYPE)
             .withAuthenticationResponses()
             .build());
+
+    this.keyManager = keyManager;
   }
 
   @Override
   public void handle(final RestApiRequest request) throws JsonProcessingException {
-    List<BLSPublicKey> keys = request.getRequestBody();
-    for (BLSPublicKey key : keys) {
-      LOG.debug("Delete key: {}", key.toBytesCompressed().toShortHexString());
+    try {
+      DeleteKeysRequest deleteRequest = request.getRequestBody();
+      if (deleteRequest == null) {
+        request.respondError(SC_BAD_REQUEST, "Could not read request body");
+        return;
+      }
+      request.respondOk(keyManager.deleteValidators(deleteRequest.getPublicKeys()));
+    } catch (JsonProcessingException ex) {
+      request.respondError(SC_BAD_REQUEST, ex.getMessage());
     }
```

Comment:
I wonder if we should be leaving these exceptions to throw and just dealing with them in a default exception handler. Not sure we have one registered for this but would be good to avoid having to have this try/catch repeated in multiple handlers.

Reply:
good point about default exception handler, will add.